jnigen/test/jackson_core_test/third_party/lib/com/fasterxml/jackson/core/JsonFactory.dart

```diff
@@ -59,7 +59,7 @@ import "../../../../_init.dart";
 /// instances.
 ///@author Tatu Saloranta
 class JsonFactory extends jni.JObject {
-  late final jni.JObjType? _$type;
+  jni.JObjType? _$type;
```

Comment:
Why is this no longer final? Does the time change over time?

Reply:
It's actually a mistake to do `somethingLate ??= something;` because `somethingLate` would have to be evaluated first to check for nullity but it's never been initialized since it's `late`. It's never changed, it's just caching `$type`.

Reply:
Correct, and you don't want it `final` because you want to initialize it at some point on demand instead of in the constructor.

I'm not sure what is a nice pattern for that. This doesn't communicate it doesn't change after it's initialized. Can't we just have an initialization expression instead?

```dart
  @override
  late final jni.JObjType $type = type;
```

Reply:
I'm not sure if that's a good idea because even if we never use `$type`, it's still going to be initialized, right?

Reply:
No, `late final`s are only initialized on demand.

(See discussion on https://dart-review.googlesource.com/c/sdk/+/286782/3..5/pkg/vm/lib/transformations/ffi/finalizable.dart#b267 )